templates/boltzmann/core/honeycomb.ts

```diff
@@ -0,0 +1,427 @@
+    this.features = {
+      honeycomb: !options.disable,
+      beeline: !options.disable,
+    }
+    this.initialized = false
+    this.started = false
+  }
+
+  public static log(message: any): void {
+    // There's a good likelihood that bole hasn't been configured yet,
+    // so we use console here. We also want honeycomb to fail gracefully
+    // as nothing is more embarrassing than your service getting taken
+    // down by instrumentation, so we only log in live dev and in debug.
+    void `{% if not selftest %}`
+    if (process.env.DEBUG || isDev()) {
+      console.warn(message);
+    }
+    void `{% endif %}`
+  }
+
+  // Initialize Honeycomb! Sets up the beeline library.
+  // This needs to occur before any imports you want instrumentation
+  // to be aware of.
+  public init(): void {
+    try {
+      const writeKey = this.getWriteKey();
+      const dataset = this.getDataset();
+      const sampleRate = this.options.sampleRate || 1;
+      const serviceName = this.options.serviceName;
+
+      if (this.features.honeycomb) {
+        beeline({ writeKey, dataset, sampleRate, serviceName })
+        this.initialized = true
+        return
+      }
+    } catch (err) {
+      if (err instanceof HoneycombError) {
+        // Honeycomb.log(err);
+        return;
+      }
+      throw err;
+    }
+  }
+
+  // When implemented, this will start the OpenTelemetry SDK. In the
+  // case of beelines, it's a no-op. In the otel case, this will need
+  // to happen before anything happens in the entrypoint and is an
+  // async operation.
+  public async start(): Promise<void> {
+    this.started = true
+  }
+
+  // Start a trace. Returns a Trace, which may be closed by calling
+  // `trace.end()`.
+  public async startTrace(context: Context, headerSources?: string[]): Promise<Span> {
+    if (!this.features.honeycomb || !this.initialized) {
+      return {
+        async end() {}
+      }
+    }
+
+    // The core of this method is hidden away in preparation for the
+    // bulk of this method being OTLP focused later.
+    return this._startBeelineTrace(context, headerSources)
+  }
+
+  // Beelines implementation for starting/ending a trace
+  private defaultHeaderSources: string[] = [ 'x-honeycomb-trace', 'x-request-id' ]
+
+  private async _startBeelineTrace(
+    context: Context,
+    headerSources?: string[]
+  ): Promise<Span> {
+    if (!this.features.honeycomb || !this.initialized) {
+      return {
+        async end() {}
+      }
+    }
+
+    const schema = require('honeycomb-beeline/lib/schema')
+    const tracker = require('honeycomb-beeline/lib/async_tracker')
+    const _headerSources = headerSources || this.defaultHeaderSources
+    const traceContext = _getBeelineTraceContext(context)
+    const trace = beeline.startTrace({
+      [schema.EVENT_TYPE]: 'boltzmann',
+      [schema.PACKAGE_VERSION]: '1.0.0',
+      [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
```

Comment:
per a discussion with ceej. It's worth verifying in your manual tests that this does the right thing, compared to what was being done before.
```suggestion
      [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}`,
```

Reply:
That change might've been an accident? 🤔 I can attempt at making this more testable though.

Reply:
Sorry, I misread this as a delta I already made, not a suggested change. I'm making it now, with a test.

Reply:
OK, change should've been made. Marking this resolved.